src/MasonryFlashList.tsx

```diff
@@ -430,8 +438,13 @@ const updateViewTokens = (tokens: ViewToken[]) => {
   for (let i = 0; i < length; i++) {
     const token = tokens[i];
     if (token.index !== null && token.index !== undefined) {
-      token.index = token.item.originalIndex;
-      token.item = token.item.originalItem;
+      if (token.item) {
+        token.index = token.item.originalIndex;
+        token.item = token.item.originalItem;
+      } else {
+        token.index = null;
+        token.item = undefined;
```

Comment:
Why `index` is set to `null` and `item` to `undefined`?

Reply:
Unfortunately, that's the type specified in ViewToken. Changing it can be a breaking change for many.